src/Data/Bool/Compat.hs

```diff
@@ -1,3 +1,6 @@
+#if MIN_VERSION_base(4,6,0) && !MIN_VERSION_base(4,7,0)
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+#endif
```

Comment:
Works for me. I got a bit carried away with the pragmas (in case you ever decided to run `hlint` on this and wanted to avoid unused language extensions). I'll fix all instances of this in other files.